modules/nf-core/blast/blastdbcmd/main.nf

```diff
@@ -0,0 +1,55 @@
+process BLAST_BLASTDBCMD {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/blast:2.15.0--pl5321h6f7f691_1':
+        'biocontainers/blast:2.15.0--pl5321h6f7f691_1' }"
+
+    input:
+    tuple val(meta) , val(entry), path(entry_batch)
+    tuple val(meta2), path(db)
+
+    output:
+    tuple val(meta), path("*.fasta"), emit: fasta
+    path "versions.yml"             , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    assert (!entry && entry_batch) || (entry && !entry_batch)
+    """
+    DB=`find -L ./ -name "*.nhr" | sed 's/\\.nhr\$//'`
+    if test -z "\$DB"
+    then
+        DB=`find -L ./ -name "*.phr" | sed 's/\\.phr\$//'`
+    fi
+    blastdbcmd \\
+        -db \$DB \\
+        ${args} \\
+        -outfmt %f \\
+        -out ${prefix}.fasta \\
+        ${entry? "-entry $entry":"-entry_batch $entry_batch"}
```

Comment:
I would put this condition in the `script:` initialization above with the other variables, and inject with a single variable, I find this a bit hard to read at the moment having to context switch between bash and groovy thinking

Reply:
Done. It didn’t work for me with a ternary operator, but it is done anyway.